core/src/main/java/net/sourceforge/jnlp/util/XDesktopEntry.java

```diff
@@ -349,16 +349,20 @@ private void cacheIcon() throws IOException, NonFileProtocolException {
              targetName.toLowerCase().endsWith(JPG)  ||
              targetName.toLowerCase().endsWith(JPEG) ||
              targetName.toLowerCase().endsWith(PNG))) {
-            target = convertToIco(cacheFile, targetName);
+            target = convertToIco(cacheFile, getPrefixedName(targetName));
         }
         if (target == null) {
-            target = new File(PathsAndFiles.ICONS_DIR.getFile(), targetName);
+            target = new File(PathsAndFiles.ICONS_DIR.getFile(), getPrefixedName(targetName));
             Files.copy(cacheFile.toPath(), target.toPath(), StandardCopyOption.REPLACE_EXISTING);
         }
         this.iconLocation = target.getAbsolutePath();
         LOG.debug("Cached desktop shortcut icon: " + target + " ,  With source from: " + cacheFile.getAbsolutePath());
     }
 
+    private String getPrefixedName(final String targetName) {
+        return file.getSourceLocation().toString().hashCode()+"-"+targetName;
```

Comment:
Should we use "-"? Also a possibility of hashCode being negative. May be we take absolute value or remove the minus sign?